dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java

```diff
@@ -163,14 +162,44 @@ public static Contentlet find(String inodeOrIdentifier, User user, String tmDate
 		public static List<Contentlet> pull(String query,int limit, String sort, User user, String tmDate){
 		    return pull(query,-1,limit,sort, user, tmDate);
 		}
-		
-		public static PaginatedArrayList<Contentlet> pull(String query, int offset,int limit, String sort, User user, String tmDate){
+
+		/**
+		 * Pull content using query if a Time Machine Date is set into session then it ios take account when the query is
+		 * ran.
+		 * 
+		 * @param query
+		 * @param offset
+		 * @param limit
+		 * @param sort
+		 * @param user
+		 * @param tmDate
+		 * @return
+		 */
+		public static PaginatedArrayList<Contentlet> pull(
+				final String query,
+				final int offset,
+				final int limit,
+				final String sort,
+				final User user,
+				final String tmDate){
 			return pull(query, offset, limit, sort, user, tmDate, PageMode.get().respectAnonPerms);
 		}
 
-		public static PaginatedArrayList<Contentlet> pull(String query, final int offset, final int limit, final String sort, final User user, final String tmDate, final boolean respectFrontendRoles){
+		public static PaginatedArrayList<Contentlet> pull(String query, final int offset, final int limit,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'query' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4740221)